### DISCUSSION TITLE
Cross compilation are failing for ARM64 platform

### DISCUSSION THREAD

**Discussion by koratshail** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1070] [Q&A] [answered]
I am trying to build a **Kinesis video streams producer sdk cpp** package for my ARM64 platform.

all attempts to build this package with the dependencies failed.
First of all, the target platform needed to be specified explicitly as shown below:
$(pkg)_CONF_OPTS       = -DBUILD_GSTREAMER_PLUGIN=ON -DBUILD_OPENSSL_PLATFORM=linux-aarch64 -DBUILD_LOG4CPLUS_HOST=arm-linux

 The problem that couldn't be solved was how to build libcurl: the compilation failed because one source **couldn't find the <openssl/opensslconf.h> include file.**
 Providing the file's location using an environment variable, as export OPENSSL_INCLUDE_DIR = $(@D)/open-source/local/include
 made the compilations pass, but failed at the linking stage, due to wrong file format, because the files were compiled for the build host platform, not the target one.

Following are the compilation log messages:
`/home/ambadocker/dev/h32/output/h32_ambalink_defconfig/build/aws_kinesis_kvs/open-source/local/libcurl/build/src/project_libcurl/lib/md4.c:31:10: fatal error: openssl/opensslconf.h: No such file or directory
   31 | #include <openssl/opensslconf.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
lib/CMakeFiles/libcurl.dir/build.make:598: recipe for target 'lib/CMakeFiles/libcurl.dir/md4.c.o' failed
make[7]: *** [lib/CMakeFiles/libcurl.dir/md4.c.o] Error 1
make[7]: Leaving directory '/home/ambadocker/dev/h32/output/h32_ambalink_defconfig/build/aws_kinesis_kvs/open-source/local/libcurl/build/src/project_libcurl-build'
CMakeFiles/Makefile2:1572: recipe for target 'lib/CMakeFiles/libcurl.dir/all' failed
make[6]: Leaving directory '/home/ambadocker/dev/h32/output/h32_ambalink_defconfig/build/aws_kinesis_kvs/open-source/local/libcurl/build/src/project_libcurl-build'
make[6]: *** [lib/CMakeFiles/libcurl.dir/all] Error 2
Makefile:143: recipe for target 'all' failed
make[5]: *** [all] Error 2
make[5]: Leaving directory '/home/ambadocker/dev/h32/output/h32_ambalink_defconfig/build/aws_kinesis_kvs/open-source/local/libcurl/build/src/project_libcurl-build'
CMakeFiles/project_libcurl.dir/build.make:115: recipe for target 'build/src/project_libcurl-stamp/project_libcurl-build' failed
make[4]: *** [build/src/project_libcurl-stamp/project_libcurl-build] Error 2
make[4]: Leaving directory '/home/ambadocker/dev/h32/output/h32_ambalink_defconfig/build/aws_kinesis_kvs/open-source/local/libcurl'
CMakeFiles/Makefile2:78: recipe for target 'CMakeFiles/project_libcurl.dir/all' failed
make[3]: *** [CMakeFiles/project_libcurl.dir/all] Error 2
make[3]: Leaving directory '/home/ambadocker/dev/h32/output/h32_ambalink_defconfig/build/aws_kinesis_kvs/open-source/local/libcurl'
Makefile:86: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/home/ambadocker/dev/h32/output/h32_ambalink_defconfig/build/aws_kinesis_kvs/open-source/local/libcurl'
CMake Error at dependency/libkvscproducer/kvscproducer-src/CMake/Utilities.cmake:93 (message):
  CMake step for libcurl failed: 2
Call Stack (most recent call first):
  dependency/libkvscproducer/kvscproducer-src/CMakeLists.txt:95 (build_dependency)`

Can somebody explain how to cross-compile this SDK for the ARM64 platform with all dependencies like curl, OpenSSL etc?

**Comment by sirknightj** [comment, marked as answer]
Duplicate of #1071.
